easybuild/easyblocks/generic/buildenv.py

```diff
@@ -60,7 +61,12 @@ def install_step(self, *args, **kwargs):
         wrappers_dir = os.path.join(self.rpath_wrappers_dir, RPATH_WRAPPERS_SUBDIR)
         if os.path.exists(wrappers_dir):
             self.rpath_wrappers_dir = os.path.join(self.installdir, 'bin')
-            copy_dir(wrappers_dir, os.path.join(self.rpath_wrappers_dir, RPATH_WRAPPERS_SUBDIR))
+            rpath_wrappers_path = os.path.join(self.rpath_wrappers_dir, RPATH_WRAPPERS_SUBDIR)
+            copy_dir(wrappers_dir, rpath_wrappers_path)
+            wrapper_files = list(filter(os.path.isfile, glob.glob(os.path.join(rpath_wrappers_path, '*', '*'))))
+            # replace path to Python executable with python3 in the wrappers,
+            # as this is the executable that runs EasyBuild and may be unavailable when using the buildenv wrappers.
+            apply_regex_substitutions(wrapper_files, [(r'^PYTHON_EXE=.*$', 'PYTHON_EXE=python3')], backup=False)
```

Comment:
Sorry to be a pain here, but `python3` is also not _guaranteed_ to exist. Can we add an easyblock `extra_options` to decide when to do this? A `None` value would mean don't do it, a `True` value would mean use the default `python3`, a string value would mean use a specific value (which may for example include a full path). This would give people the power (via hooks) to decide what is best for them.